lib/rouge/lexers/mxml.rb

```diff
@@ -0,0 +1,244 @@
+
+      state :root do
+        mixin :xml_root
+      end
+
+      state :xml_root do
+        rule /[^<&]+/, Text
+        rule /&\S*?;/, Name::Entity
+        rule /<!\[CDATA\[/, Comment::Preproc, :actionscript
+        rule /<!--/, Comment, :xml_comment
+        rule /<\?.*?\?>/, Comment::Preproc
+        rule /<![^>]*>/, Comment::Preproc
+
+        # open tags
+        rule %r(<\s*[\w:.-]+)m, Name::Tag, :xml_tag
+
+        # self-closing tags
+        rule %r(<\s*/\s*[\w:.-]+\s*>)m, Name::Tag
+      end
+
+      state :xml_comment do
+        rule /[^-]+/m, Comment
+        rule /-->/, Comment, :xml_root
+        rule /-/, Comment
+      end
+
+      state :xml_tag do
+        rule /\s+/m, Text
+        rule /[\w.:-]+\s*=/m, Name::Attribute, :xml_attribute
+        rule %r(/?\s*>), Name::Tag, :xml_root
+      end
+
+      state :xml_attribute do
+        rule /\s+/m, Text
+        rule /"{|"@{/, Str, :actionscript
+        rule /".*?"|'.*?'|[^\s>]+/, Str, :xml_tag
+      end
+
+      state :actionscript do
+        rule /\]\]\>/ do
+          token Comment::Preproc
+          goto :xml_root
+        end
+        
+        rule /\}"/ do
+          token Str
+          goto :xml_tag
+        end
+        
+        mixin :actionscript_root
+      end
+      
+      def self.keywords
+        @keywords ||= Set.new %w(
+          for in while do break return continue switch case default
+          if else throw try catch finally new delete typeof is
+          this with
+        )
+      end
+
+      def self.declarations
+        @declarations ||= Set.new %w(var with function)
+      end
+
+      def self.reserved
+        @reserved ||= Set.new %w(
+          dynamic final internal native public protected private class const
+          override static package interface extends implements namespace
+          set get import include super flash_proxy object_proxy trace
+        )
+      end
+
+      def self.constants
+        @constants ||= Set.new %w(true false null NaN Infinity undefined)
+      end
+
+      def self.builtins
+        @builtins ||= %w(
+          void Function Math Class
+          Object RegExp decodeURI
+          decodeURIComponent encodeURI encodeURIComponent
+          eval isFinite isNaN parseFloat parseInt this
+        )
+      end
+
+      id = /[$a-zA-Z_][a-zA-Z0-9_]*/
+      
+      state :actionscript_root do
+        rule /\A\s*#!.*?\n/m, Comment::Preproc, :actionscript_statement
+        rule /\n/, Text, :actionscript_statement
+        rule %r((?<=\n)(?=\s|/|<!--)), Text, :actionscript_expr_start
+        mixin :actionscript_comments_and_whitespace
+        rule %r(\+\+ | -- | ~ | && | \|\| | \\(?=\n) | << | >>>? | ===
+               | !== )x,
+          Operator, :actionscript_expr_start
+        rule %r([:-<>+*%&|\^/!=]=?), Operator, :actionscript_expr_start
+        rule /[(\[,]/, Punctuation, :actionscript_expr_start
+        rule /;/, Punctuation, :actionscript_statement
+        rule /[)\].]/, Punctuation
```

Comment:
Instead of reimplementing the ActionScript lexer here, please delegate to the existing one.  See the HTML lexer and how it handles `<script>` and `<style>` tags for an example.